bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_3drepo.cpp

```diff
@@ -285,15 +287,43 @@ RepoModelImport::mesh_data_t RepoModelImport::createMeshRecord(
 
 			uint32_t *tmpIndices = (uint32_t*)(dataBuffer + startEnd[0]);
 
-			for (int i = 0; i < numIndices; i += 3)
+			// figure out primitive type
+			boost::optional<const ptree&> primitiveProp = mesh.get_child_optional(REPO_IMPORT_PRIMITIVE);
+			MeshNode::Primitive primitiveType;
+			if (primitiveProp)
 			{
-				repo_face_t tmpFace;
-
-				tmpFace.push_back(tmpIndices[i]);
-				tmpFace.push_back(tmpIndices[i + 1]);
-				tmpFace.push_back(tmpIndices[i + 2]);
+				primitiveType = static_cast<MeshNode::Primitive>(primitiveProp.get().get_value<int8_t>());
+			}
+			else
+			{
+				primitiveType = MeshNode::Primitive::TRIANGLES;
+			}
 
-				faces.push_back(tmpFace);
+			// pull out faces
+			switch (primitiveType)
+			{
+			case repo::core::model::MeshNode::Primitive::LINES:
+				for (int i = 0; i < numIndices; i += 2)
+				{
+					repo_face_t tmpFace;
+					tmpFace.push_back(tmpIndices[i]);
+					tmpFace.push_back(tmpIndices[i + 1]);
+					faces.push_back(tmpFace);
+				}
+				break;
+			case repo::core::model::MeshNode::Primitive::TRIANGLES:
+				for (int i = 0; i < numIndices; i += 3)
+				{
+					repo_face_t tmpFace;
+					tmpFace.push_back(tmpIndices[i]);
+					tmpFace.push_back(tmpIndices[i + 1]);
+					tmpFace.push_back(tmpIndices[i + 2]);
+					faces.push_back(tmpFace);
+				}
+				break;
+			default:
+				//TODO: how to error out here?
```

Comment:
@sebjf  added a new error code for identifying import errors regarding unexpected geometry content:  REPOERR_ODA_GEOMETRY_ERROR

Should probably rename to REPOERR_GEOMETRY_ERROR and use that.